evm/p2p/lightchain.py

```diff
@@ -106,10 +103,13 @@ def register_peer(self, peer: BasePeer) -> None:
         await peer.stop()
 
     async def wait_until_finished(self):
+        count = 0
         while self._running_peers:
```

Comment:
What do you think about this.

```python
for _ in range(50):
    if not self._running_peers:
        break
    self.logger.debug("Waiting for %d running peers to finish", len(self._running_peers))
    await asyncio.sleep(0.1)
else:
    self.logger.info("Waited too long for peers to finish, exiting anyway")
```

Reply:
Alternate to remove the *weird* magic number `50`

```python
start_at = time.time()
while time.time() < start_at + 5:
    ...
else:
    ...
```

Makes it a bit more explicit how long the shutdown process is allowed to wait for.

Reply:
Went with the latter